gridpath/transmission/operations/operational_types/tx_simple.py

```diff
@@ -327,6 +373,32 @@ def max_losses_to_rule(mod, l, tmp):
             * mod.Tx_Availability_Derate[l, tmp] \
             * mod.tx_simple_loss_factor[l]
 
+
+def min_transmit_power_rule(mod, l, bt, h):
+    """
+    **Constraint Name**: TxSimple_Min_Transmit_Power_Constraint
+    **Enforced Over**: TX_SIMPLE_OPR_TMPS_W_MIN_CONSTRAINT
+
+    Transmitted power should exceed the defined minimum transmission power in
+    each operational timepoint.
+
+    If the minimum transmitted power is positive, the transmitted power should
+    exceed the minimum transmitted power in the transmission line's positive direction.
+
+    If the minimum transmitted power is negative, the transmitted power should
+    exceed the minimum transmitted power in transmission line's negative direction.
+    """
+    var = mod.tx_simple_min_transmit_power_mw[l, bt, h]
+    if var == 0:
+        return Constraint.Skip
+    elif var > 0:
+        for tmp in mod.TMPS_BY_BLN_TYPE_HRZ[bt, h]:
```

Comment:
Instead of iterating over the timepoints in a balancing_type-horizon in the constraint rule definition, I would suggest creating the equivalent set `(project, bh, h, tmp)` and enforcing the constraint over that set. At the end, this is a timepoint-level constraint and you want to be able to see what timepoints it binds in (right now, you would only see the balancing_type-horizon when the constraint binds, so you are losing resolution).

Related to this is whether to even define the min and max flow parameters over a balancing_type-horizon or if it's better to just define them by timepoint. I would recommend the latter as it gives us more flexibility. Defining a balancing_type-horizon level parameter makes sense if we would be aggregating over that balancing_type-horizon, which here we are not. There trade-offs of course: in one case we need to define the parameter over a smaller set (balancing_type-horizons instead of timepoints) but we do need to create that set; in the other case, it's more cumbersome to define the parameter over each timepoint (as it's potentially the same over all those timepoints) but we do not need to worry about defining the balancing_type-horizons in the temporal structure.

Reply:
@anamileva I just reopened this code to try to finish it. 

Regarding your first comment on having two parameters tx_simple_min_flow_mw and tx_simple_max_flow_mw and two constraints, I think how you propose to define the two constraints can be confusing. For example, if we want to impose a minimum flow in the negative direction of the transmission line, we would need to put a negative value for the tx_simple_max_flow_mw parameter, so that TxSimple_Transmit_Power_MW<= tx_simple_max_flow_mw.

As to defining the parameters over timepoints instead of balancing_type-horizon, I agree. I will do the changes.